json-rpc/tests/integration_test.rs

```diff
@@ -339,6 +339,45 @@ fn create_test_cases() -> Vec<Test> {
                 );
             },
         },
+        Test {
+            name: "get account by version",
+            run: |env: &mut testing::Env| {
+                let account = &env.vasps[0];
+                let address = format!("{:x}", &account.address);
+                // The account txn is creating child VASP.
+                let account_sequence = 0;
+
+                let resp = env.send("get_account_transaction", json!([address, account_sequence, false]));
+                let result = resp.result.unwrap();
+                let prev_version: u64 = result["version"].as_u64().unwrap() - 1;
+                let resp = env.send("get_account", json!([address, prev_version]));
+                let result = resp.result.unwrap();
+                assert_eq!(
+                    result,
+                    json!({
+                        "address": address,
+                        "authentication_key": account.auth_key().to_string(),
+                        "balances": [{"amount": 1000000000000_u64, "currency": "XUS"}],
+                        "delegated_key_rotation_capability": false,
+                        "delegated_withdrawal_capability": false,
+                        "is_frozen": false,
+                        "received_events_key": format!("0200000000000000{}", address),
+                        "role": {
+                            "base_url": "",
+                            "base_url_rotation_events_key": format!("0100000000000000{}", address),
+                            "compliance_key": "",
+                            "compliance_key_rotation_events_key": format!("0000000000000000{}", address),
+                            "expiration_time": 18446744073709551615_u64,
+                            "human_name": "Novi 0",
+                            "num_children": 0,
+                            "type": "parent_vasp"
+                        },
+                        "sent_events_key": format!("0300000000000000{}", address),
+                        "sequence_number": 0
+                    }),
```

Comment:
this test doesn't seem to be too useful as there's nothing here about version afaiu, don't we really just care that the version in the result is correctly set? also it seems like the json blob should be a bunch of predefined fields but things like expiration date, human name, and balances are not constants.

Reply:
the test first find transaction version of account sequence number, then use the transaction version - 1 to find back the older version account. This ensures we can find back older version by version param.

the test is running with a single node with db included, it is hard to find out what is version the account is created or before last change.

the test above, tests same account and get back latest account info.
the difference between them are balances and sequence_number.

Reply:
the test first finds the first transaction version of the account (account sequence number 0),
then use the (version - 1) as transaction version to look up account, which gives the account data when it is set up.
Mostly to ensure we can look up previous version.
I'm not sure what you mean by "be a bunch of predefined fields but things like expiration date, human name, and balances are not constants.", these data all set up by test env.

the response version is in header or response body, every response contains it. but it maybe nice to have in Account response body too, we can add.